mail/api.py

```diff
@@ -377,15 +378,15 @@ def get_mail_vars(emails):
         generator of dict:
             A dictionary of template variables which includes email so we can tell who is who
     """
-    queryset = User.objects.filter(email__in=emails).values(
-        'email',
-        'profile__mail_id',
-        'profile__preferred_name',
+    queryset = Profile.objects.filter(user__email__in=emails).values(
+        'user__email',
+        'mail_id',
+        'preferred_name',
```

Comment:
This a pretty large query considering that it needs to join all the profiles with the users.
What about querying the users with the emails and then filtering the profiles by those users?